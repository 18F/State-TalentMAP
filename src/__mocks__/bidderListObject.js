const bidderListObject = {
  count: 3,
  results: [
    {
      id: 1,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Doe',
        email: 'doej@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 2,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'Jane',
        last_name: 'Doe',
        email: 'doejane@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
    {
      id: 3,
      skill_code: 'INFORMATION MANAGEMENT (2880)',
      user: {
        first_name: 'John',
        last_name: 'Smith',
        email: 'smithj@state.gov',
      },
      is_cdo: false,
      bid_statistics: [
        {
          draft: 1,
          submitted: 1,
        },
      ],
    },
  ],
};

export default bidderListObject;
