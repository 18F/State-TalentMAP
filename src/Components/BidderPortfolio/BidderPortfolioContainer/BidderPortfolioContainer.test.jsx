import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import toJSON from 'enzyme-to-json';
import BidderPortfolioContainer from './BidderPortfolioContainer';
import bidderListObject from '../../../__mocks__/bidderListObject';

describe('BidderPortfolioContainerComponent', () => {
  it('is defined', () => {
    const wrapper = shallow(<BidderPortfolioContainer
      bidderPortfolio={bidderListObject}
      pageSize={8}
      pageNumber={1}
      queryParamUpdate={() => {}}
    />);
    expect(wrapper).toBeDefined();
  });

  it('can call the onPageChange function', () => {
    const spy = sinon.spy();
    const wrapper = shallow(<BidderPortfolioContainer
      bidderPortfolio={bidderListObject}
      pageSize={8}
      pageNumber={1}
      queryParamUpdate={spy}
    />);
    wrapper.instance().onPageChange({});
    sinon.assert.calledOnce(spy);
  });

  it('matches snapshot when there are results', () => {
    const wrapper = shallow(<BidderPortfolioContainer
      bidderPortfolio={bidderListObject}
      pageSize={8}
      pageNumber={1}
      queryParamUpdate={() => {}}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when there are no results', () => {
    const wrapper = shallow(<BidderPortfolioContainer
      bidderPortfolio={Object.assign({}, bidderListObject, { count: 0, results: [] })}
      pageSize={8}
      pageNumber={1}
      queryParamUpdate={() => {}}
    />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
