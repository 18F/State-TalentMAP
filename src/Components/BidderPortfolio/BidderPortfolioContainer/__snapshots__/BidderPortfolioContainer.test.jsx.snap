// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BidderPortfolioContainerComponent matches snapshot when the all property is greater than zero 1`] = `
<div
  className="usa-grid-full user-dashboard"
>
  <BidderPortfolioCardList
    results={
      Array [
        Object {
          "id": 1,
          "is_cdo": false,
          "skill_code": "INFORMATION MANAGEMENT (2880)",
          "user": Object {
            "email": "doej@state.gov",
            "first_name": "John",
            "last_name": "Doe",
          },
        },
        Object {
          "id": 2,
          "is_cdo": false,
          "skill_code": "INFORMATION MANAGEMENT (2880)",
          "user": Object {
            "email": "doejane@state.gov",
            "first_name": "Jane",
            "last_name": "Doe",
          },
        },
        Object {
          "id": 3,
          "is_cdo": false,
          "skill_code": "INFORMATION MANAGEMENT (2880)",
          "user": Object {
            "email": "smithj@state.gov",
            "first_name": "John",
            "last_name": "Smith",
          },
        },
      ]
    }
  />
  <div
    className="usa-grid-full react-paginate"
  >
    <PaginationWrapper
      activeClassName="active"
      containerClassName="pagination"
      forcePage={1}
      marginPagesDisplayed={2}
      nextLabel="Next"
      onPageChange={[Function]}
      pageRangeDisplayed={1}
      pageSize={8}
      paginationStyle="pagination"
      previousLabel="Previous"
      subContainerClassName="pages pagination"
      totalResults={3}
    />
  </div>
</div>
`;

exports[`BidderPortfolioContainerComponent matches snapshot when the all property is zero 1`] = `
<div
  className="usa-grid-full user-dashboard"
>
  <BidderPortfolioCardList
    results={Array []}
  />
  <div
    className="usa-width-two-thirds"
  >
    <Alert
      messages={
        Array [
          Object {
            "body": "Try removing filters or using another bid status tab.",
          },
        ]
      }
      title="You have no clients within this search criteria."
      type="info"
    />
  </div>
</div>
`;
